terraform/environments/analytical-platform-compute/s3-buckets.tf

```diff
@@ -21,3 +21,72 @@ module "mlflow_bucket" {
 
   tags = local.tags
 }
+
+module "mojap_derived_tables_replication_bucket" {
+  #checkov:skip=CKV_TF_1:Module registry does not support commit hashes for versions
+  #checkov:skip=CKV_TF_2:Module registry does not support tags for versions
+
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "4.2.1"
+
+  bucket = "mojap-derived-tables-replication-${local.environment}"
+
+  force_destroy = true
+
+  object_lock_enabled = false
+
+  acl = "private"
+
+  versioning = {
+    status = "Disabled"
+  }
+
+  server_side_encryption_configuration = {
+    rule = {
+      bucket_key_enabled = true
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = module.mojap_derived_tables_replication_s3_kms.key_arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+
+  logging = {
+    target_bucket = module.apc_bucket_logs.s3_bucket_id
+    target_prefix = "mojap-derived-tables-replication/"
+  }
+
+  tags = local.tags
+}
+
+module "apc_bucket_logs" {
```

Comment:
This bucket needs a bucket policy to enable log delivery:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html#grant-log-delivery-permissions-general <-- the section you need is enabling log delivery via bucket policy